src/aggregate/aggregate_request.c

```diff
@@ -334,8 +334,13 @@ static int handleCommonArgs(AREQ *req, ArgsCursor *ac, QueryError *status, int a
     req->prefixesOffset = ac->offset - 1;
 
     // Advance by the number of prefixes + 1 (for the n_prefixes argument)
-    long advance_by = strtol((const char *)AC_CURRENT(ac), NULL, 10);
-    AC_AdvanceBy(ac, advance_by + 1);
+    uint32_t advance_by;
+    if (AC_GetUnsigned(ac, &advance_by, AC_F_GE1) != AC_OK) {
```

Comment:
why not call `AC_GetVarArgs` to both advance and validate the number + that we have enough arguments to advance by?

Reply:
I don't need the copying mechanism to another ArgsCursor.
Added another assertion, as this case should never happen.

Reply:
I don't need the copying mechanism, and I also want to have assertions for this, as this should never happen